core/commands/add.go

```diff
@@ -113,12 +113,16 @@ remains to be implemented.
 					return
 				}
 
-				err = n.Pinning.Pin(context.Background(), rootnd, true)
+				rnk, err := rootnd.Key()
 				if err != nil {
 					res.SetError(err, cmds.ErrNormal)
 					return
 				}
 
+				mp := n.Pinning.GetManual()
+				mp.RemovePinWithMode(rnk, pin.Indirect)
```

Comment:
> Coupling that with something like Git's gc.pruneexpire 

that's a great solution. of course it comes from git.

i'm really concerned about safety here-- 

@whyrusleeping @tv42 -- what's the plan long term?